myocamlbuild.ml

```diff
@@ -32,7 +32,7 @@ let () =
         ["link"; "ocaml"; "link_mtime_os_stubs"] (P "src-os/libmtime_stubs.a");
 
       flag ["library"; "ocaml"; "byte"; "record_mtime_os_stubs"]
-        (S ([A "-dllib"; A "-lmtime_stubs"] @ system_support_lib));
+        (S ([A "-cclib"; A "-lmtime_stubs"] @ [A "-dllib"; A "dllmtime_stubs.so"] @ system_support_lib));
```

Comment:
This is certainly not the right fix 1) the `-cclib` you add is already added by the next rule 2) You should not have to specify the object file explicitely (for one thing the `.so` extension is not portable).